h2o-genmodel/src/main/java/hex/genmodel/algos/targetencoder/TargetEncoderMojoModel.java

```diff
@@ -0,0 +1,77 @@
+package hex.genmodel.algos.targetencoder;
+
+import hex.genmodel.MojoModel;
+import hex.genmodel.easy.RowData;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class TargetEncoderMojoModel extends MojoModel implements FeatureTransformerModel{
+
+  public TargetEncoderMojoModel(String[] columns, String[][] domains, String responseName) {
+    super(columns, domains, responseName);
+  }
+
+  public EncodingMaps _targetEncodingMap;
+  public boolean _withBlending;
+  public double _inflectionPoint;
+  public double _smoothing;
+
+  // Could be passed from the model training phase so that we don't have to recompute it here. Or maybe it is fine as it should be computed only once per mojoModel
+  private double _priorMean = -1; 
+
+  @Override
+  public RowData transform(RowData data) {
+    
+    if(_targetEncodingMap != null) {
+      for (Map.Entry<String, EncodingMap> columnToEncodingsMap : _targetEncodingMap.entrySet()) {
+        String columnName = columnToEncodingsMap.getKey();
+        String originalValue = (String) data.get(columnName);
+        EncodingMap encodings = columnToEncodingsMap.getValue();
+
+        _priorMean = _priorMean == -1 ? computePriorMean(encodings) : _priorMean;
+        
+        int[] correspondingNumAndDen = encodings._encodingMap.get(originalValue);
+
+        int numberOfRowsInCurrentCategory = correspondingNumAndDen[1];
```

Comment:
This question is too-o-o general :) Let's say in TE I always do imputation with NA category... otherwise our `merge` function would fail to merge it properly.